scheduler/resource/task_manager.go

```diff
@@ -104,11 +104,21 @@ func (t *taskManager) RunGC() error {
 		task := value.(*Task)
 		elapsed := time.Since(task.UpdateAt.Load())
 
-		if elapsed > t.ttl && task.PeerCount() == 0 && !task.FSM.Is(TaskStateRunning) {
+		if elapsed > t.ttl && task.FSM.Is(TaskStateLeave) {
 			task.Log.Info("task has been reclaimed")
 			t.Delete(task.ID)
+			return true
 		}
 
+		// If there is no peer then switch the task state to TaskStateLeave.
+		if task.PeerCount() == 0 {
+			if err := task.FSM.Event(TaskEventLeave); err != nil {
```

Comment:
Why trigger `TaskEventLeave` here? Is it better to trigger `TaskEventLeave` when grpc of TaskLeave is called?

Reply:
Unified logic in the GC is clear.

Reply:
But GC is called periodically with a long interval(like 1min). It is not timely.

Reply:
If it is not timely, there is no problem in letting peers go back-to-source directly. Actively triggering the seed peer is also going back-to-source.